src/quicklist.c

```diff
@@ -469,26 +475,34 @@ static size_t quicklistNodeNegFillLimit(int fill) {
 
 /* Calculate the size limit or length limit of the quicklist node
  * based on 'fill', and is also used to limit list listpack. */
-void quicklistNodeLimit(int fill, size_t *size, unsigned int *count) {
+REDIS_STATIC void _quicklistNodeLimit(int fill, size_t *size,
+                                      unsigned int *count,
+                                      size_t size_limit) {
     *size = SIZE_MAX;
     *count = UINT_MAX;
 
     if (fill >= 0) {
         /* Ensure that one node have at least one entry */
         *count = (fill == 0) ? 1 : fill;
     } else {
-        *size = quicklistNodeNegFillLimit(fill);
+        *size = (size_limit? size_limit : quicklistNodeNegFillLimit(fill));
```

Comment:
```suggestion
        assert(size_limit > 0);
        *size = size_limit;
```
when fill < 0, size_limit should not be less than 0.

Reply:
size_limit is size_t (unsigned), so it can't be less than 0. Even if in quicklistCreate, I had initialized it to a valid value. As for why I need to determine whether it is 0 here, it is because I need to support the wrapper mentioned above.